app.rb

```diff
@@ -0,0 +1,169 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require 'date'
+
+class Main
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @menu_options = {
+      1 => method(:list_books),
+      2 => method(:list_people),
+      3 => method(:create_person),
+      4 => method(:create_book),
+      5 => method(:create_rental),
+      6 => method(:list_rentals_for_person),
+      7 => method(:quit_app)
+    }
+  end
+
+  def list_books
+    puts 'Listing all books:'
+    @books.each { |book| puts "#{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    puts 'Listing all people:'
+    @people.each { |person| puts "#{person.class}: #{person.name} (ID: #{person.id})" }
+  end
+
+  def create_person
+    puts 'Is the person a student or a teacher? (S/T)'
+    person_type = gets.chomp.upcase
+
+    if person_type == 'S'
+      create_student

```

Comment:
- [ ] Please ensure to implement the `app.rb` as stated in the reqirement, all these methods (**`from line 24 -123, 158-168`**) supposed to be in `app.rb` file. main.rb should only be defined as the entry point, with a method that is invoked at the end of your file. ℹ️ You can reference this as a [refresher](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point).

<img width="698" alt="Untitled" src="https://github.com/uno36/OOP-School-Library-ruby/assets/103386183/e9468b2f-4ff0-4a88-a3fa-c772cf7fb194">